src/olmo_core/train/trainer.py

```diff
@@ -627,7 +641,11 @@ def fit(self):
             # Then fallback to the load path, if provided.
             if self.load_path is not None:
                 if not self.checkpoint_loaded:
-                    self.maybe_load_checkpoint(self.load_path)
+                    self.maybe_load_checkpoint(
+                        self.load_path,
+                        load_trainer_state=self.load_trainer_state,
+                        load_optim_state=self.load_optim_state,
```

Comment:
I think it would make more sense to default to these settings within the `.load_checkpoint()` method. That way any call to `.load_checkpoint/.maybe_load_checkpoint` will respect the settings.